orchagent/flexcounterorch.cpp

```diff
@@ -69,12 +72,22 @@ unordered_map<string, string> flexCounterGroupMap =
 
 FlexCounterOrch::FlexCounterOrch(DBConnector *db, vector<string> &tableNames):
     Orch(db, tableNames),
-    m_flexCounterConfigTable(db, CFG_FLEX_COUNTER_TABLE_NAME),
     m_bufferQueueConfigTable(db, CFG_BUFFER_QUEUE_TABLE_NAME),
     m_bufferPgConfigTable(db, CFG_BUFFER_PG_TABLE_NAME),
     m_deviceMetadataConfigTable(db, CFG_DEVICE_METADATA_TABLE_NAME)
 {
     SWSS_LOG_ENTER();
+    m_delayTimer = new SelectableTimer(timespec{.tv_sec = FLEX_COUNTER_DELAY_SEC, .tv_nsec = 0});
```

Comment:
>new [](http://example.com/codeflow?start=19&length=3)

where do you delete it? in destructor?

Reply:
will you use smart pointer?

Reply:
@qiluo-msft I am ok using smart pointer. Did not choose to in this situation due to the lifetime of FlexCounterOrch and therefore m_delayTimer is the entire duration of the program, therefore no resource leak. Looking at other usages it seems to me that we aren't really calling destructor in other places as well, e.g. https://github.com/sonic-net/sonic-swss/blob/master/orchagent/portsorch.cpp#L538